torch/csrc/autograd/engine.cpp

```diff
@@ -352,7 +379,7 @@ static void validate_outputs(const edge_list& edges, variable_list& grads, const
       ss << metadata.type() << " but got " << grads[i].type();
       AT_ERROR(format_error(ss.str()));
     }
-    const auto output_device = output.is_cuda() ? output.get_device() : -1;
+    const int output_device = assign_tensor_to_autograd_thread(output);
```

Comment:
Well that seems weird. If you have a function called `assign_tensor_to_autograd_thread`, then it feels weird to use it to validate if the output is on a correct device. Someone will change this function in the future, and will not expect that they should look into `validate_outputs` as well...

Reply:
Well I'd really like to see this one fixed... Can't we just compare `.device()` of two tensors here? I just don't want to use a function that assigns threads to select the device.